### DISCUSSION TITLE
Use Livewire with nwidart/modules

### DISCUSSION THREAD

**Discussion by woganmay** [livewire/livewire#1813] [3. Feature ideas]
I maintain a fairly complex application (relying on nwidart/modules) and would like to use Livewire for some of the modules, but right now everything in Livewire is designed to only support one namespace at a time. When creating the components and moving them around I can get them to work within the modules, but it's not an ideal solution.

From what I can see, there's only really 2 changes needed to support this:

1. The ability to specify a root namespace per component at creation time (defaulting to the config), and
2. The ability to customize the list of namespaces and paths that the LivewireComponentsFinder scans.

For the first, it seems that everything rests on where the ComponentParser says the files should go. If it were possible to inject my own parser object into that command, I could probably achieve that without any further modifications to the existing code.

For the second, I implemented a quick-and-dirty solution that overrides the getClassNames() method on the finder, but otherwise writes to the usual manifest path. This works perfectly once the discovery runs: https://gist.github.com/woganmay/52ae70b058c22fb41e66b27c2a1d45ee

I tried binding that implementation via the container, but it looks like I can't override the singleton that Livewire is creating for it. If it were possible to do that, I think I'd be halfway there already.

What would the preferred approach be, here? I can submit a PR to implement this, and try to have it affect as little of the existing code as possible.

**Comment by adrian-ub** [comment]
You can try this [Package Development](https://laravel-livewire.com/docs/2.x/package-dev) 
It's the way I'm doing it with the package [InterNACHI/Modular](https://github.com/InterNACHI/modular)

**Comment by Azmeer-crescent** [comment]
My problem is, the Livewire components work ok on localhost, but doesn't work on the live server. My app runs on a subfolder in the live server. I get 404 for all routes within Modules, eg:
`https://<server>/<folder>/livewire/message/<module name>::<component name>`
